### PR TITLE
Skills create controller routes

### PR DESCRIPTION
**Pull request by briwagner**
Hey Will. Please take a look when you can.

Couple things going on here. I'll admit I haven't written Express routes in a while, but the Sequelize try-catch syntax is really throwing me. It seems really verbose, too many lines for what it's doing. I tried something a bit different, see what you think. No hard feelings if you don't think it works. Ultimately, we should have one pattern for all of these routes, for simplicity and consistency. I hesitated on the "delete" route; it seems delete returns [] when it succeeds and that could be the way to handle errors. Not sure ...

Separately, my sense is we are returning JSON on all of these routes. So I'm doing that here. That may be something we change on the other routes, which are returning normal body content currently.

Also error status. We probably want to return status codes on some of these error cases. We don't expect the frontend to parse error strings, right?

Docker-compose: I goofed and forgot to expose ports. That could be a separate PR if you want. I just added here.

### CODE REVIEW

**Review by raindogg — COMMENTED** [review]
Thanks Brian! Definitely not married to `try catch`, and the way you've done here is easier to read, so we can go with that. I'll make a PR to update to this style (plus using `res.json()` and `findByPk`, which are also nice timesavers). I left a couple comments as well.

File: db/controllers/skills.controller.js [hunk 1]
```diff
@@ -0,0 +1,72 @@
+const { seq } = require('../index.js');
```
[the remark this answers is not in the document]
Comment (briwagner) [reply db/controllers/skills.controller.js:1]: Good catch. I copied from the existing files and started to modify. Didn't look at this.

File: db/controllers/skills.controller.js [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+const { models } = require('../index');
+
+exports.getSkills = async (req, res) => {
+  var dbError
+  const skills = await models.skill.findAll()
+      .catch(err => dbError = err);
+
+  if (dbError) { return res.status(400).json({"error": dbError}); }
```
[the remark this answers is not in the document]
Comment (briwagner) [reply db/controllers/skills.controller.js:9]: Yeah, it doesn't properly return the response that way. The way the promise works.

Here's the context where I was debating the alternatives: https://stackoverflow.com/questions/40884153/try-catch-blocks-with-async-await/48527368#48527368

File: db/controllers/skills.controller.js [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+        return res.sendStatus(404);
+    }
+
+    const {
+        name,
+        description
+    } = req.body;
+
+    var dbError;
+    skill.update({name: name, description: description})
```
[the remark this answers is not in the document]
Comment (briwagner) [reply db/controllers/skills.controller.js:51]: I separated them so we can clearly handle the different error cases. 404 vs. 400 for some DB error.

Looking at this, it seems sequelize doesn't give us the model back, for an update. Assuming we want to return the model in the response.

More info and some options here: https://newbedev.com/sequelize-update
IMO, this doesn't look much cleaner; seems like you're still doing two requests, just chaining them together.

**Review by raindogg — APPROVED** [review]

**Pull request merged**